public-ui/src/app/data-browser/charts/chart-survey-answers/react-survey-answer-chart.component.tsx

```diff
@@ -0,0 +1,275 @@
+    array.sort((a, b) =>
+     order.indexOf(a[key]) - order.indexOf(b[key])
+    );
+  }
+
+  public conceptDist(sortedAnswers: any) {
+    let tempArr: any[] = [];
+    const categoryArr = [];
+    const { answerChartInfo } = this.state;
+    for (const prop in sortedAnswers) {
+      if (sortedAnswers.hasOwnProperty(prop)) {
+        categoryArr.push(prop);
+        const answerOrder = answerChartInfo.map(p => p.answerId);
+        this.mapOrder(sortedAnswers[prop], answerOrder, 'stratum3');
+        sortedAnswers[prop].forEach(answer => {
+          tempArr.push({
+            name: answer.stratum4,
+            data: []
+          });
+        });
+      }
+    }
+
+    // remove duplicates
+    tempArr =
+      Array.from(new Set(tempArr.map(x => JSON.stringify(x)))).map(x => JSON.parse(x));
+
+    for (const prop in sortedAnswers) {
+      if (sortedAnswers.hasOwnProperty(prop)) {
+        tempArr.forEach(stack => {
+          sortedAnswers[prop].forEach(answer => {
+            if (stack.name === answer.stratum4) {
+              stack.data.push(answer.countValue);
+            }
+          });
+        });
+      }
+    }
+    this.setState({chartSeries: tempArr, categoryArr: categoryArr}, () => {
+        this.buildChartData();
+    });
+  }
+
+  buildChartData() {
+    const { categoryArr, chartSeries, colors } = this.state;
+    const newBaseOptions = getBaseOptions();
+    newBaseOptions.chart.type = 'column';
+    newBaseOptions.xAxis.categories = categoryArr;
+    newBaseOptions.series = chartSeries;
+    newBaseOptions.yAxis.title.text = 'Participant Count';
+    newBaseOptions.yAxis.title.style.fontSize = '16px';
+    newBaseOptions.xAxis.labels.style.fontSize = '14px';
+    newBaseOptions.yAxis.labels.style.fontSize = '16px';
+    newBaseOptions.yAxis.title.style.padding = '1rem';
+    newBaseOptions.plotOptions.column.stacking = 'normal';
+    newBaseOptions.plotOptions.column.colorByPoint = false;
+    newBaseOptions.plotOptions.column.pointWidth = 50;
+    newBaseOptions.plotOptions.column.borderWidth = 0;
+    newBaseOptions.plotOptions.series.animation = {
+        duration: 0
+    };
+    newBaseOptions.plotOptions.series.fontSize = '14px';
+    newBaseOptions.legend = {
+          enabled: false
+    };
+    newBaseOptions.tooltip.followPointer = true;
+    newBaseOptions.tooltip.useHTML = true;
+    newBaseOptions.tooltip.backgroundColor = 'transparent';
+    newBaseOptions.tooltip.formatter = function() {
+        const count = (this.point.y <= 20) ? '&le; 20' : this.point.y;
+        const percentage = ((count / this.point.total) * 100).toFixed();
+        this.point.toolTipHelpText = `
+            <div class="survey-answer-tooltip">
+            <strong>${this.point.series.name}</strong>
+            <span>${count} Participants </span>
+            <span>${percentage}% of all participants who took this version of survey</span>
+            <span>${this.point.total} Total </div></span>`;
```

Comment:
Misplaced closing `</div>` / `</span>` and indent nested elements:
```suggestion
            <div class="survey-answer-tooltip">
              <strong>${this.point.series.name}</strong>
              <span>${count} Participants </span>
              <span>${percentage}% of all participants who took this version of survey</span>
              <span>${this.point.total} Total </span>
            </div>`
```

Reply:
Changd thanks.